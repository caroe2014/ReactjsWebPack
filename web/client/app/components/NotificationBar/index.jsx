// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Drawer, Flex } from 'rebass';
import PrimaryButton from 'web/client/app/components/PrimaryButton';
import Announcements from 'web/client/app/components/Announcements';

const NotificationContainer = styled(Drawer)`
  display: flex;
  background-color: rgba(0,0,0,.7);
  border-radius: 10px;
  color: ${props => props.theme.colors.light};
  box-shadow: -1px 0 2px rgba(0,0,0,0.5);
  margin-bottom: 35px;
  margin-left: auto;
  margin-right: auto;
  min-width: 400px;
  max-width: 800px;
  &:hover {
    box-shadow: none;
  }
  ${props => props.theme.mediaBreakpoints.mobile`
    min-width: 300px;
  `}
`;

const MainButton = styled(props => <PrimaryButton {...props} />)`
  border-radius: 4px;
  font-size: 0.8rem;
  font-weight: 600;
  min-width: 50px;
  padding: 4px;
  text-align: center;
  margin: 0;
  margin-left: auto;
  background: none;
  width: auto;
  white-space: nowrap;
  color: ${props => props.theme.colors.light};
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

class NotificationBar extends Component {

  constructor (props) {
    super(props);
    this.closeNotificationBar = this.closeNotificationBar.bind(this);
  }

  componentDidUpdate (prevProps) {
    if (!prevProps.open && this.props.open) {
      this.closeNotificationBar();
    }
  }

  closeNotificationBar () {
    if (this.props.open) {
      setTimeout(
        function () {
          this.props.onActionClick();
        }.bind(this),
        4000
      );
    }
  }

  render () {
    const { open, text, actionButtonText, onActionClick } = this.props;
    return (
      <Flex>
        {open &&
        <NotificationContainer
          role='alert'
          open={open}
          side='bottom'
          p={3}>
          <Announcements message={text} ariaLive='assertive'/>
          <span tabIndex={0} style={{outline: 'none', boxShadow: 'none'}}
            aria-label={text}>{text}</span>
          <MainButton
            children={actionButtonText}
            onClick={onActionClick}
            tabIndex={0}
            aria-label={actionButtonText}
            role='button'
          />
        </NotificationContainer>
        }
      </Flex>
    );
  }
}

NotificationBar.propTypes = {
  open: PropTypes.bool,
  actionButtonText: PropTypes.object,
  onActionClick: PropTypes.func,
  text: PropTypes.string
};

export default NotificationBar;
