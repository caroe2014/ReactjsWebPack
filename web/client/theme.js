// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
