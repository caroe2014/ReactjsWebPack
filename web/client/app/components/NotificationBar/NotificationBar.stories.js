// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs/react';
import { Provider } from 'rebass';
import theme from 'web/client/theme.js';
import NotificationBar from '../NotificationBar';

storiesOf('Panels', module)
  .addDecorator(withKnobs)
  .add('NotificationBar', () => {
    return (
      <div>
        <Provider theme={theme}>
          <NotificationBar title='Panel Header'>
            <p>Panel Content</p>
          </NotificationBar>
        </Provider>
      </div>
    );
  })
  .add('NotificationBar With Selection', () => {
    return (
      <div>
        <Provider theme={theme}>
          <NotificationBar title='Panel Header' selection='Selection Text'>
            <p>Panel Content</p>
          </NotificationBar>
        </Provider>
      </div>
    );
  });
