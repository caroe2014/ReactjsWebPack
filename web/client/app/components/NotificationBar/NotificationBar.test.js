// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import {
  shallow,
  mount,
  configure
} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {
  ThemeProvider
} from 'styled-components';
import theme from 'web/client/theme';

import NotificationBar from './index';
configure({
  adapter: new Adapter()
});

describe('NotificationBar', () => {

  it('does a snapshot test', () => {
    const wrapper = shallow(<NotificationBar open/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(shallow(<NotificationBar open/>).exists()).toBe(true);
  });

  it('should render without throwing an error on mount', () => {
    expect(mount(<ThemeProvider theme={theme}><NotificationBar open/></ThemeProvider>).exists()).toBe(true);
  });
});
