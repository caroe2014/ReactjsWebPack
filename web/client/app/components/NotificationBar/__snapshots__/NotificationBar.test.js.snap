// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationBar does a snapshot test 1`] = `
<Flex>
  <Styled(Drawer)
    open={true}
    p={3}
    role="alert"
    side="bottom"
  >
    <Announcements
      ariaLive="assertive"
    />
    <span
      style={
        Object {
          "boxShadow": "none",
          "outline": "none",
        }
      }
      tabIndex={0}
    />
    <Styled(Component)
      role="button"
      tabIndex={0}
    />
  </Styled(Drawer)>
</Flex>
`;
